src/api/api.ts

```diff
@@ -1,69 +1,175 @@
-import { Body, fetch as tauriFetch } from '@tauri-apps/api/http';
+import { Body, fetch as tauriFetch, ResponseType } from '@tauri-apps/api/http';
 
 import { REFSTUDIO_HOST } from './server';
 
+type HttpMethod = 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH';
+type ResponseParser = 'JSON' | 'ArrayBuffer';
+
 /** Issue a GET request using either web fetch or Tauri fetch */
-export async function universalGet<ResponsePayload = unknown>(path: string): Promise<ResponsePayload> {
-  if (import.meta.env.VITE_IS_WEB) {
-    const response = await fetch(path);
-    if (!response.ok) {
-      throw new Error(`Failed to get ${path}: ${response.statusText}`);
-    }
-    return response.json() as ResponsePayload;
-  } else {
-    const response = await tauriFetch(REFSTUDIO_HOST + path);
-    if (!response.ok) {
-      throw new Error(`Failed to get ${path}: ${response.status}`);
-    }
-    return response.data as ResponsePayload;
-  }
+export async function universalGet<ResponsePayload = unknown>(
```

Comment:
Two things here:

1. `unknown` is the "default default" for type parameters in TS, so there's no need to write "= unknown" explicitly.
2. This is an example of a so-called [return-only generic](https://effectivetypescript.com/2020/08/12/generics-golden-rule/). These are generally frowned upon in TS because they don't actually give you any type safety -- you can claim the return type is whatever you want. The better way would be to have this always return `Promise<unknown>` and then do `await universalGet('...') as ResponsePayload` in the calling code since at least that way it's explicit that you're doing a type assertion.

That being said, I think this will all change when we plug in code to generate these types so fine to leave as-is for now.

Reply:
Ok, thanks for sharing this information. I leave as-is for now bc we will update this soon.